react/features/recording/components/Recording/AbstractRecordButton.js

```diff
@@ -0,0 +1,138 @@
+// @flow
+
+import {
+    createToolbarEvent,
+    sendAnalytics
+} from '../../../analytics';
+
+import { openDialog } from '../../../base/dialog';
+import { JitsiRecordingConstants } from '../../../base/lib-jitsi-meet';
+import {
+    getLocalParticipant,
+    isLocalParticipantModerator
+} from '../../../base/participants';
+import {
+    AbstractButton,
+    type AbstractButtonProps
+} from '../../../base/toolbox';
+
+import { getActiveSession } from '../../functions';
+
+import StartRecordingDialog from './StartRecordingDialog';
+import StopRecordingDialog from './StopRecordingDialog';
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link AbstractRecordButton}.
+ */
+export type Props = AbstractButtonProps & {
+
+    /**
+     * True if there is a running active recording, false otherwise.
+     */
+    _isRecordingRunning: boolean,
+
+    /**
+     * The redux {@code dispatch} function.
+     */
+    dispatch: Function,
+
+    /**
+     * The i18n translate function.
+     */
+    t: Function
+};
+
+/**
+ * An abstract implementation of a button for starting and stopping recording.
+ */
+export default class AbstractRecordButton<P: Props>
+    extends AbstractButton<P, *> {
```

Comment:
I think this is how we format, and eslint warns me if I do otherwise

![image](https://user-images.githubusercontent.com/9974975/42164199-91192f2c-7e05-11e8-8bd3-ea95784d1d97.png)

Reply:
Ack. It somehow doesn't complain on `if`s, oh well!

Reply:
Right. Btw I'm about to add some new eslint rules, maybe those will help here